src/App.tsx

```diff
@@ -1,8 +1,72 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Button, Container, Header } from 'semantic-ui-react';
+import { PostList } from './components/PostList';
+import { Search } from './components/Search';
+import { getPosts } from './api/getPosts';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of posts</h1>
-);
+const App = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [isLoaded, setLoaded] = useState(false);
+  const [isLoading, setLoading] = useState(false);
+  const [isError, setError] = useState('');
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const loadData = async () => {
+    setLoading(true);
+
+    try {
+      const data = await getPosts();
+
+      setPosts(data);
+      setLoading(false);
+      setLoaded(true);
+    } catch (error) {
+      setLoading(false);
+      setError(`Something went wrong! ${error.message}`);
+    }
+  };
+
+  const searchPosts = () => {
+    if (!searchQuery) {
+      return posts;
+    }
+
+    return posts.filter(({ title, body }) => (`${title} ${body}`)
+      .toLowerCase()
+      .includes(searchQuery));
+  };
+
+  const searchedPosts = searchPosts();
```

Comment:
add useMemo for optimization

Reply:
![Screenshot_1](https://user-images.githubusercontent.com/36817641/82360196-b0d39800-9a11-11ea-904f-0d4b28a9315e.png)
![Screenshot_2](https://user-images.githubusercontent.com/36817641/82360234-bfba4a80-9a11-11ea-9a53-bcdb0486e62e.png)